src/repository/opamGit.ml

```diff
@@ -216,7 +214,12 @@ module VCS : OpamVCS.VCS = struct
                    @ List.map OpamFilename.SubPath.to_string
                      (OpamStd.Option.to_list subpath))
     @@> function
-    | { OpamProcess.r_code = 0; _ } -> Done true
+    | { OpamProcess.r_code = 0; _ } ->
+      git repo_root ["submodule"; "status"; "--recursive"] @@> fun r ->
+      if r.r_code = 0 &&
+         List.for_all (fun s -> String.length s > 0 && s.[0] = ' ') r.r_stdout
```

Comment:
I haven't checked the full context of how this function is used, but is an uninitialised submodule here possible - is this used for pinning?

Reply:
It is called transitively from each `OpamRepository.pull_*` functions.